$$\color{MediumSeaGreen}{\text{NOTEBOOK III ESTADÍSTICA INFERENCIAL}}$$

$$\color{MediumSeaGreen}{\text{DIPLOMADO TÉCNICAS ESTADISTICAS Y MINERÍA DE DATOS}}$$

$$\color{MediumSeaGreen}{\text{MODULÓ II: MODELOS ESTADÍSTICOS}}$$

$$\color{MediumSeaGreen}{\text{FES ACATLÁN, UNAM}}$$

---

<div align="center">

<img src="https://unade.edu.mx/wp-content/uploads/2025/09/estadistica_descriptiva.jpg" width="80%">

</div>

</center>

$$\color{MediumSeaGreen}{\text{Nombre:Allison Ximena Fragoso Velázquez}}$$

$$\color{MediumSeaGreen}{\text{Profesor: Act. Claudia Cristina Reyes Montes de Oca}}$$

$$\color{MediumSeaGreen}{\text{Fecha: 27 de julio de 2026}}$$

# $\color{OliveDrab}{\text{Estimadores}}$

Un estimador es una función de los datos muestrales utilizada para aproximar un parámetro poblacional desconocido.

Ejemplos:

- $\bar X$ estima a $\mu$:

 La media muestral
$$\bar X=\frac{1}{n}\sum_{i=1}^{n}X_i$$

- $S^2$ estima a $\sigma^2$.
- $\hat p$ estima a $p$.

El objetivo de un estimador es proporcionar una aproximación del parámetro poblacional utilizando únicamente la información disponible en la muestra.



## Método de momentos

El método de momentos consiste en igualar los momentos muestrales con los momentos teóricos de una distribución para obtener una estimación de sus parámetros.

Por ejemplo, para una distribución exponencial:

$$
E(X)=\frac1{\lambda}
$$

Si la media muestral es $\bar X$, el estimador es

$$
\hat\lambda=\frac1{\bar X}
$$


In [2]:
import numpy as np
import scipy.stats as st
import matplotlib.pyplot as plt

muestra = np.array([2.1, 0.7, 1.5, 3.0, 0.9, 1.2])
lambda_momentos = 1/muestra.mean()

print("Media muestral:", muestra.mean())
print("Estimación de lambda por momentos:", lambda_momentos)


Media muestral: 1.5666666666666664
Estimación de lambda por momentos: 0.6382978723404257


#$\color{MediumSeaGreen}{\text{Interpretación}}$

La media muestral se utiliza para estimar el parámetro $\lambda$ Conforme aumenta el tamaño de la muestra, la estimación obtenida mediante el método de momentos suele aproximarse al valor verdadero del parámetro.


## Estimación por máxima verosimilitud (EMV)

La EMV selecciona el valor del parámetro que hace más probable la muestra observada.

Para la distribución exponencial:

$$
\hat\lambda_{EMV}=\frac{1}{\bar X}
$$

En este ejemplo coincide con el método de momentos.


In [3]:
lambda_emv = 1/muestra.mean()
print("Estimación de lambda por EMV:", lambda_emv)


Estimación de lambda por EMV: 0.6382978723404257


#$\color{MediumSeaGreen}{\text{Interpretación}}$

La estimación por máxima verosimilitud selecciona el valor del parámetro que maximiza la probabilidad de obtener los datos observados. En este ejemplo, el valor estimado es cercano al parámetro utilizado para generar la muestra.

# $\color{OliveDrab}{\text{Propiedades de los estimadores}}$

## Insesgadez

Un estimador es **insesgado** cuando, en promedio, coincide con el verdadero valor del parámetro.

Matemáticamente,

$$
E(\hat\theta)=\theta
$$

**Interpretación**

Si se tomaran muchas muestras, el promedio de todas las estimaciones sería igual al parámetro poblacional.

En promedio, el estimador no sobrestima ni subestima el parámetro.


## Consistencia

Un estimador es **consistente** cuando se aproxima al parámetro verdadero conforme aumenta el tamaño de la muestra.

$$
\hat\theta\rightarrow\theta
$$

**Interpretación**

Mientras más información se tenga, menor será el error de estimación.

## Eficiencia

Un estimador es **eficiente** cuando presenta la menor varianza posible entre todos los estimadores insesgados.

**Interpretación**

Dos estimadores pueden ser insesgados, pero el más eficiente produce estimaciones menos dispersas y, por tanto, más precisas.


In [4]:
# Ejemplo visual de consistencia de la media
np.random.seed(123)
mu = 10

for n in [5, 20, 100, 1000]:
    muestra_n = np.random.normal(mu, 2, n)
    print(f"n={n:4d} | media estimada={muestra_n.mean():.3f}")


n=   5 | media estimada=9.244
n=  20 | media estimada=10.539
n= 100 | media estimada=10.002
n=1000 | media estimada=9.919


# $\color{OliveDrab}{\text{Intervalos de confianza}}$

Un intervalo de confianza es un rango de valores que con cierta probabilidad (nivel de confianza) contiene al verdadero parámetro poblacional que queremos estimar.

Sea $X_1,\dots, X_n$ una muestra aleatoria con $X\sim\text{N}(\mu,\sigma^2)$, entonces
$$(num_1 < \mu < num2)$$

En los intervalos de confianza, tenemos el error estándar (SE) el cual mide cuanto varía un estimador.
 - si conocemos $\sigma$, entonces: $SE = \sigma/\sqrt{n}$
 - si NO conocemos $\sigma$, utilizamos $s$ y entonces: $SE = s/\sqrt{n

NOTA: a mayor tamaño de la muestra se tiene que el SE será menor y por lo tanto el intervalo de confianza será más preciso.

**Def** Si $X_1,\dots, X_n$ es una muestra aleatoria con densidad $f(x;\theta)$ y sean $T_1=t_1(X_1,\dots, X_n)$ y $T_2=t_2(X_1,\dots, X_n)$ dos estadísticos talque $T_1 \le T_2$ entonces
$$\mathbb{P}(T_1 \le \theta \le T_2) = \gamma$$
Al intervalo $(T_1,T_2)$ se le llama intervalo del $\gamma\times 100\%$ de confianza.


In [5]:
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as st

## 1. Intervalo de confianza para la media de una población normal

Sea $X_1,\dots, X_n$ una muestra aleatoria con $X\sim\text{N}(\mu,\sigma^2)$, entonces
 - $\bar{X_i} \sim \mathcal{N}(\mu, \sigma^2/n)$
 - $Z = \frac{\bar{X_i}-\mu}{\sigma/n} \sim \mathcal{N}(0,1)$

### $\sigma$ conocida:

Se tiene que el intervalo del $(1-\alpha)\times 100\%$ de confianza para $\mu$ es:
$$ \bar{X}\pm z_{\alpha/2}\frac{\sigma}{\sqrt{n}}$$
donde
 - $z_{\alpha/2}\frac{\sigma}{\sqrt{n}}$ es el margen de error
 - $z_{\alpha/2}$ es el valor crítico

**Ejemplo 1** Para tratar de estimar la media de consumo por cliente en un restaurante, se reunieron datos de una muestra de tamaño $49$ clientes durante tres semanas.

**a)** Supongamos que la desviación estándar poblacional es de $\$2.5$. ¿Cuál es el error estandar de la media?

**b)** Con un nivel de confianza del $95\%$, ¿cuál es el margen de error?

**c)** Si la media muestral es de $\$22.60$, cual es el intervalo de confianza del $95\%$ para la media poblacional.

In [6]:
# Datos:
n = 49 # número de clientes
sigma = 2.5
media_muestral = 22.60
confianza = 0.95

# a) SE
error_est = sigma / np.sqrt(n)
print('El error estándar es:', error_est)

# b) 1-confianza = 0.95 -> 0.025 en cada cola -> (1-confianza)/2
z = st.norm.ppf(1-(1-confianza)/2)
margen_error = z*error_est
print('El margen de error es:', margen_error)

# c) Intervalo de confianza
limite_inf = media_muestral - margen_error
limite_sup = media_muestral + margen_error
print(f'El intervalo del {confianza*100}% es: ({limite_inf:.2f},{limite_sup:.2f})')

El error estándar es: 0.35714285714285715
El margen de error es: 0.6999871373357336
El intervalo del 95.0% es: (21.90,23.30)


Por lo tanto, se tiene que el verdadero consumo poblacional en el restaurante se encuentra entre $\$21.90$ y $\$23.30$

### $\sigma$ desconocida:

Se tiene que el intervalo del $(1-\alpha)\times 100\%$ de confianza para $\mu$ es:
$$ \bar{X}\pm t_{\alpha/2}\frac{\sigma}{\sqrt{n}}$$
donde
 - $t_{\alpha/2}\frac{\sigma}{\sqrt{n}}$ es el margen de error
 - $t_{\alpha/2}$ es el valor crítico

**Ejemplo 2**  Supongamos que tenemos los siguientes datos de una muestra aleatoria de 10 observaciones:
$$45,55,67,68,45,79,98,87,84,82$$

Calcular un intervalo de confianza para la media.

In [7]:
# Datos
datos = [45,55,67,68,45,79,98,87,84,82] # n<= 30
n = len(datos)
confianza = 0.99
gl = n-1

media_muestral = np.mean(datos)
error_est = st.sem(datos) # eror estandar muestral

# Vamos a utilizar la función
# scipy.stats.t.interval(confidence, df, loc, scale)
# confidence = confianza
# df = degree freedom (gl)
# loc = media_muestral
# scale = error_est

intervalo = st.t.interval(confianza, gl, media_muestral, error_est)
print(f'El intervalo de confianza del {confianza*100}%: ({intervalo[0]:.2f},{intervalo[1]:.2f})')

El intervalo de confianza del 99.0%: (52.32,89.68)


#$\color{MediumSeaGreen}{\text{Interpretación}}$

Con un **nivel de confianza del 99 %**, se estima que la media poblacional se encuentra entre **52.32** y **89.68** (o los valores que obtengas al ejecutar el código).

Esto significa que, si se repitiera el proceso de tomar muestras del mismo tamaño y construir un intervalo de confianza para cada una, aproximadamente el **99 % de esos intervalos contendrían la verdadera media poblacional**.

En este ejemplo se utiliza la **distribución t de Student** porque el tamaño de la muestra es pequeño $(n=10)$ y la desviación estándar de la población es desconocida.

## 2. Intervalo de confianza para la varianza de una población normal

### $\mu$ desconocida:

Se tiene que el intervalo del $(1-\alpha)\times 100\%$ de confianza para $\sigma^2$ es:
$$ \left( \frac{(n-1)s^2}{\chi^{2}_{1-\alpha/2}}, \frac{(n-1)s^2}{\chi^{2}_{\alpha/2}} \right)$$

**Ejemplo 3** Los artículos de cerámica utilizados sobre velas electricas se rompen con diferentes presiones. Supongamos que los datos provienen de una distribución normal. La resistencia a la ruptura fue medida en una muestra de 100 artículos y el promedio fue de $1750$ con una desviación estándar de $315.8$.

**a)** Estimar con un nivel de confianza del $90\%$ a la media poblacional de la presión de la ruptura.

**b)** Estimar con un nivel de confianza del $90\%$ a la varianza poblacional.

In [8]:
# Datos
n = 100
confianza = 0.90
media_muestral = 1750
desv_est = 315.8
error_est = desv_est / np.sqrt(n)

# a) como n>=30, utilizo la normal
intervalo = st.norm.interval(confianza, media_muestral, error_est)
print(f'El intervalo de confianza del {confianza*100}%: ({intervalo[0]:.2f},{intervalo[1]:.2f})')

# b) IC para la varianza
gl = n-1
alpha = 1-confianza
s2 = desv_est**2 # varianza muestral

# Forma 1. Calculando los cuatiles  de la chi cuadrada
chi_inf = st.chi2.ppf(alpha/2,gl)
chi_sup = st.chi2.ppf(1-alpha/2,gl)
int_var = ((gl*s2) / chi_sup, (gl*s2) / chi_inf)
print(f'El intervalo de confianza del {confianza*100}%: ({int_var[0]:.2f},{int_var[1]:.2f})')

# Forma 2.
chi2_inf, chi2_sup = st.chi2.interval(confianza, gl)
int2_var = ((gl*s2) / chi2_sup, (gl*s2) / chi2_inf)
print(f'El intervalo de confianza del {confianza*100}%: ({int2_var[0]:.2f},{int2_var[1]:.2f})')


El intervalo de confianza del 90.0%: (1698.06,1801.94)
El intervalo de confianza del 90.0%: (80123.49,128146.72)
El intervalo de confianza del 90.0%: (80123.49,128146.72)


In [9]:
chi_inf, chi_sup

(np.float64(77.04633186376029), np.float64(123.2252214533618))

#$\color{MediumSeaGreen}{\text{Interpretación}}$

### a) Intervalo de confianza para la media

Con un **nivel de confianza del 90 %**, se estima que la **media poblacional de la presión de ruptura** de los artículos de cerámica se encuentra dentro del intervalo calculado por la distribución normal.

En este caso se utiliza la **distribución normal** porque el tamaño de la muestra es grande $(n=100)$, lo que permite aplicar el Teorema Central del Límite. Esto significa que, si se repitiera el proceso de muestreo muchas veces y se construyeran intervalos de confianza del 90 %, aproximadamente el **90 % de ellos contendrían la verdadera media poblacional**.

---

### b) Intervalo de confianza para la varianza

El intervalo obtenido estima el rango en el que se encuentra la **varianza poblacional** con un nivel de confianza del **90 %**.

Para este cálculo se utiliza la **distribución ji-cuadrada $\chi^2$**, ya que los datos provienen de una población con distribución normal. El intervalo proporciona valores plausibles para la variabilidad de la resistencia a la ruptura de los artículos de cerámica. Un intervalo más amplio indica mayor incertidumbre en la estimación, mientras que un intervalo más estrecho refleja una estimación más precisa.


# Complemento: intervalo para una proporción

Cuando se desea estimar una proporción poblacional, el intervalo de confianza está dado por

$$
\hat p\pm z_{\alpha/2}\sqrt{\frac{\hat p(1-\hat p)}{n}}
$$


In [10]:
# Ejemplo: 62 respuestas favorables de 100
x = 62
n = 100
p_hat = x/n
z = st.norm.ppf(0.975)
error = np.sqrt(p_hat*(1-p_hat)/n)

li = p_hat-z*error
ls = p_hat+z*error

print("Proporción estimada:", p_hat)
print("IC del 95%:", (li, ls))


Proporción estimada: 0.62
IC del 95%: (np.float64(0.5248660051214322), np.float64(0.7151339948785678))


#$\color{MediumSeaGreen}{\text{Interpretación}}$

Con un 95% de confianza, la proporción poblacional de respuestas favorables se encuentra dentro del intervalo calculado. El intervalo muestra la incertidumbre asociada a la estimación muestral.


# $\color{MediumSeaGreen}{\text{Comparación de intervalos de confianza}}$

Los intervalos de confianza permiten estimar parámetros poblacionales a partir de una muestra. La elección de la distribución depende del tipo de parámetro, el tamaño de muestra y la información disponible sobre la población.

La distribución **normal** se utiliza cuando existe suficiente información sobre la población o el tamaño de muestra es grande. La distribución **t de Student** se utiliza cuando la muestra es pequeña y existe mayor incertidumbre debido a que la desviación estándar poblacional es desconocida. La distribución **ji-cuadrada** permite estimar la varianza, mientras que la aproximación normal permite construir intervalos para proporciones.

En todos los casos, un intervalo más estrecho indica una estimación más precisa, mientras que un intervalo más amplio refleja mayor incertidumbre en la estimación.

# $\color{OliveDrab}{\text{Conclusión}}$

En este notebook se estudiaron los principales conceptos de la **estadística inferencial**, los cuales permiten obtener información sobre una población a partir de una muestra. Se analizaron los estimadores y dos de los métodos más utilizados para estimar parámetros poblacionales: el **método de momentos** y la **estimación por máxima verosimilitud**, comprendiendo cómo se utilizan para aproximar valores desconocidos de una población.

Asimismo, se revisaron las propiedades que debe cumplir un buen estimador, como la **insesgadez**, la **consistencia** y la **eficiencia**, ya que estas permiten evaluar la calidad de las estimaciones obtenidas.

Finalmente, se construyeron **intervalos de confianza** para la media, la proporción y la varianza utilizando las distribuciones **normal**, **t de Student** y **ji-cuadrada**, según las características de cada problema. Los ejemplos desarrollados y las implementaciones en Python permitieron comparar los resultados teóricos con los cálculos computacionales, facilitando la interpretación de los intervalos y mostrando cómo la estadística inferencial constituye una herramienta fundamental para cuantificar la incertidumbre y apoyar la toma de decisiones basada en datos.

#$\color{MediumSeaGreen}{\text{Bibliografía}}$

- Rincón, L. *Introducción a la probabilidad*.
- Ross, S. *A First Course in Probability*.
- Mood, A., Graybill, F. y Boes, D. *Introduction to the Theory of Statistics*.
- Mendenhall, W. *Introduction to Probability and Statistics*.
- Material y notebooks proporcionados en clase.
